### DISCUSSION TITLE
Support data links in logs panel created in panel config

### DISCUSSION THREAD

**Discussion by chancez** [grafana/grafana#45684] [Observability]
**What would you like to be added**:

I'd like to have the ability to go from a log in a log's panel to other dashboards using [data links](https://grafana.com/docs/grafana/latest/linking/data-links/).

**Why is this needed**:

To improve the ability to go from a set of logs, to related dashboards for a given log entry. 

For example, when using Loki you might have a logQL query that shows only warnings/errors for a particular application, and you might want to go to a related metrics dashboard for the same time period, filtered down to the specific instance of the application based on the labels on the log being passed to the dashboard via url variables.

**Comment by MeetzhDing** [comment]
How is the progress of this issue?

Why provide derived field function for loki data source separately instead of directly creating data link in logs panel?

@zoltanbedi @ivanahuckova

**Comment by ivanahuckova** [comment]
>Why provide derived field function for loki data source separately instead of directly creating data link in logs panel?

Derived fields are very useful for users that use Explore for Loki querying and who wants to use derived fields to (for example) jump to traces from logs.

**Comment by siddjoshi** [comment]
Is it in plan to make data links (not Derived fields links) available in logs panel ? It can be very helpful when our log source is not Loki and still we would like to drilldown on particulars of a log entry.

**Comment by axymen** [comment]
I have dashboard with logs panel and a lot of options for our QA team. They can easily search logs for app and environment by choosing it from the drop down list without using LogQL. In the queries for logs panel I have regexp or json parser for extract labels. So, how I can add link to label traceId? I can't use derived fields  function because of our logs types. We have json based, and java based logs. Derived fields only support regexp filter. I can't use json here. I didn't find any solution for configuring link to our tempo from logs panel. Could you help me?

**Comment by ivanahuckova** [comment]
@axymen I suggest you ask on our [community site](https://community.grafana.com/) or [community slack](grafana.slack.com).
